src/indexedarray.jl

```diff
@@ -251,3 +236,51 @@ function _getcache(sa::IndexedVarArray{N,T}, pat::P) where {N,T,P}
     end
     return sa.index_cache[t]
 end
+
+function _from_linear(i, d)
+    r = Int[]
+    for dx in d
+        push!(r, mod1(i, dx))
+        i = div(i + dx - 1, dx)
+    end
+    return tuple(r...)
+end
+
+function _linear_lookup(sa::IndexedVarArray, i::Integer)
+    active = _from_linear(i, [length(i) for i in sa.index_names])
+    return [sa.index_names[i][active[i]] for i in 1:length(active)]
+end
+
+# Size and length over full range (as for SparseArray)
+Base.size(A::IndexedVarArray) = tuple([length(i) for i in A.index_names]...)
+Base.length(sa::IndexedVarArray) = prod(size(sa))
+nnz(sa::IndexedVarArray) = length(_data(sa))
+
+# Linear lookup (not sure how useful this is, but mandated by interface for AbstractArray)
+function Base.getindex(sa::IndexedVarArray{N,T}, i::Integer) where {N,T}
+    return Base.getindex(sa, _linear_lookup(sa, i)...)
+end
+
+# Extension for standard JuMP macros
+function Containers.container(
```

Comment:
I seem to get to the fallback function when using the container=IndexedVarArray, i.e. I always get the default index names.

Reply:
Yes, you need this JuMP PR: https://github.com/jump-dev/JuMP.jl/pull/3088